lib/ansible/modules/network/cloudengine/ce_snmp_user.py

```diff
@@ -0,0 +1,1074 @@
+        conf_str = kwargs["conf_str"]
+
+        xml_str = set_nc_config(module, conf_str)
+
+        return xml_str
+
+    def check_snmp_v3_usm_user_args(self, **kwargs):
+        """ Check snmp v3 usm user invalid args """
+
+        module = kwargs["module"]
+        result = dict()
+        result["usm_user_info"] = []
+        need_cfg = False
+        state = module.params['state']
+        usm_user_name = module.params['usm_user_name']
+        remote_engine_id = module.params['remote_engine_id']
+
+        acl_number = module.params['acl_number']
+        user_group = module.params['user_group']
+        auth_protocol = module.params['auth_protocol']
+        auth_key = module.params['auth_key']
+        priv_protocol = module.params['priv_protocol']
+        priv_key = module.params['priv_key']
+
+        local_user_name = module.params['aaa_local_user']
+
+        if usm_user_name:
+
+            if local_user_name:
+                module.fail_json(
+                    msg='Error: Please do not input usm_user_name and local_user_name at the same time.')
+
+            if len(usm_user_name) > 32 or len(usm_user_name) == 0:
+                module.fail_json(
+                    msg='Error: The length of usm_user_name %s is out of [1 - 32].' % usm_user_name)
+            if remote_engine_id:
+                if len(remote_engine_id) > 64 or len(remote_engine_id) < 10:
+                    module.fail_json(
+                        msg='Error: The length of remote_engine_id %s is out of [10 - 64].' % remote_engine_id)
+
+            conf_str = CE_GET_SNMP_V3_USM_USER_HEADER
+
+            if acl_number:
+                if acl_number.isdigit():
+                    if int(acl_number) > 2999 or int(acl_number) < 2000:
+                        module.fail_json(
+                            msg='Error: The value of acl_number %s is out of [2000 - 2999].' % acl_number)
+                else:
+                    if not acl_number[0].isalpha() or len(acl_number) > 32 or len(acl_number) < 1:
+                        module.fail_json(
+                            msg='Error: The length of acl_number %s is out of [1 - 32].' % acl_number)
+
+                conf_str += "<aclNumber></aclNumber>"
+
+            if user_group:
+                if len(user_group) > 32 or len(user_group) == 0:
+                    module.fail_json(
+                        msg='Error: The length of user_group %s is out of [1 - 32].' % user_group)
+
+                conf_str += "<groupName></groupName>"
+
+            if auth_protocol:
+                conf_str += "<authProtocol></authProtocol>"
+
+            if auth_key:
+                if len(auth_key) > 255 or len(auth_key) == 0:
+                    module.fail_json(
+                        msg='Error: The length of auth_key %s is out of [1 - 255].' % auth_key)
+
+                conf_str += "<authKey></authKey>"
+
+            if priv_protocol:
+                if not auth_protocol:
```

Comment:
@gundalow I need your help. when I deal this logic:
```
if A:
    if not B:
        print something errror
```
B is not dependent on A, but A dependent on B， They don't need to be config at the same time， it's not use ```required_together```, so I'll try:
```
required_if=[('B', not None, ['A'])]
```
but it is not work well.
I think ```required_if``` needs to be improved, This condition is not enough to judge when the condition is ```not None```,  How do you think about?